### PR TITLE
Add golangci-lint and fix the issues

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: aisuko <urakiny@gmail.com>

Signed off: yes
fixes: #82

### CODE REVIEW

File: pkg/cluster/cluster_test.go [hunk 1]
```diff
@@ -164,32 +162,45 @@ func TestClusterApplyDockerDesktopStartClusterOnly(t *testing.T) {
...
-		Product: string(ProductDockerDesktop),
-		MinCPUs: 0,
-	})
+	f, _ = controllerApply(f, ProductDockerDesktop, 0)
 	assert.Equal(t, true, f.d4m.started)
 	assert.Equal(t, 1, f.dockerClient.ncpu)
 }
 
+func controllerApply(f *fixture, product Product, cpus int) (*fixture, error) {
+	// We should to know that if the cpu <0 ,it's mean that we use the default value of CPU but may not equals to 0.
```
Comment (Aisuko) [inline pkg/cluster/cluster_test.go:171]: I'm not sure the logic is good. I mean may I misunderstand something here. So, please check here.

File: pkg/cluster/cluster_test.go [hunk 2]
```diff
@@ -472,7 +483,7 @@ type fakeAdmin struct {
 	deleted         *api.Cluster
 	config          *clientcmdapi.Config
 	fakeK8s         *fake.Clientset
-	serverVersion   *version.Info
+	//serverVersion   *version.Info
```
Comment (Aisuko) [inline pkg/cluster/cluster_test.go:486]: Is good we remove this field? In fact, we did not use this field.

**Comment by nicks** [comment]
woooo! this is great thank you!

I don't understand why CircleCI isn't running this. I think I messed up the configuration so that it doesn't run for PRs. Hold on...

**Comment by nicks** [comment]
ok, i think i know what the issue is -- I'm going to merge this and fix it up afterwards. Thanks again @Aisuko !

File: Makefile [hunk 3]
```diff
@@ -15,3 +17,9 @@ tidy:
 e2e:
 	test/e2e.sh
 
+.PHONY: golangci-lint
+golangci-lint: $(GOLANGCILINT)
+	$(GOPATH)/bin/golangci-lint run --verbose
+
+$(GOLANGCILINT):
```
Comment (nicks) [inline Makefile:24]: how do you run this rule? I haven't seen this makefile convention before

Comment (Aisuko) [reply Makefile:24]: `make golangci-lint`

**Pull request merged**